builder/processor/src/main/java/io/helidon/builder/processor/FactoryMethods.java

```diff
@@ -0,0 +1,327 @@
+                        .filter(it -> createMethod.equals(it.elementName()))
+                        .filter(it -> it.parameterArguments().size() == 1)
+                        .filter(it -> CONFIG_TYPE.equals(it.parameterArguments().get(0).typeName()))
+                        .filter(it -> typeInfo.typeName().equals(it.typeName()))
+                        .findFirst()
+                        .map(it -> new FactoryMethod(typeInfo.typeName(), typeInfo.typeName(), createMethod, CONFIG_TYPE));
+                if (foundMethod.isPresent()) {
+                    return foundMethod;
+                }
+            }
+        }
+
+        for (TypeInfo typeInfo : candidates) {
+            // if the target type implements ConfiguredType, we use the generic parameter of that interface to look for our config
+            // look for "implements ConfiguredType"
+
+            if (doesImplement(typeInfo, RUNTIME_OBJECT_TYPE)) {
+                // there is no config factory method available for the type that we have
+                TypeName candidateTypeName = typeInfo.typeName();
+                // we are now interested in a method with signature "static T create(Config)" where T is the type we are handling
+                Optional<FactoryMethod> foundMethod = typeInfo
+                        .elementInfo()
+                        .stream()
+                        .filter(TypeInfoPredicates::isMethod)
+                        .filter(TypeInfoPredicates::isStatic)
+                        .filter(it -> createMethod.equals(it.elementName()))
+                        .filter(it -> it.parameterArguments().size() == 1)
+                        .filter(it -> CONFIG_TYPE.equals(it.parameterArguments().get(0).typeName()))
+                        .filter(it -> candidateTypeName.equals(it.typeName()))
+                        .findFirst()
+                        .map(it -> new FactoryMethod(candidateTypeName, candidateTypeName, createMethod, CONFIG_TYPE));
+                if (foundMethod.isPresent()) {
+                    return foundMethod;
+                }
+            }
+        }
+
+        // TODO when in the same module, we need to use different approach through definition
+        // for now, just guess
+        for (TypeName configObjectCandidate : configObjectCandidates) {
+            if (configObjectCandidate.packageName().isEmpty()) {
+                // most likely a generated type that is created as part of this round, let's assume it is a config object
+                return Optional.of(new FactoryMethod(configObjectCandidate, configObjectCandidate, "create", CONFIG_TYPE));
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    private static boolean doesImplement(TypeInfo typeInfo, TypeName interfaceType) {
+        return typeInfo.interfaceTypeInfo()
+                .stream()
+                .anyMatch(it -> interfaceType.equals(it.typeName().genericTypeName()));
```

Comment:
you need to check super type info also. suggest that you make this package private and write a unit test.

Reply:
no I don't, this is checking a single level on purpose, as the iteration over supertypes/implemented interfaces hapens on the call site